File: PERSISTENCIA/EJERCICIO-1/ArchivoEmpleado.py
import pickle
import os
from Empleado import Empleado

class ArchivoEmpleado:
    def __init__(self, nomA: str):
        self.nomA = nomA

    def crearArchivo(self):
        with open(self.nomA, "wb") as archivo:
            pickle.dump([],archivo)
    
    def guardarEmpleado(self, e: Empleado):
        empleados = []
        if os.path.exists(self.nomA):
            with open(self.nomA, "rb") as archivo:
                empleados = pickle.load(archivo)
        empleados.append(e)
        with open(self.nomA, "wb") as archivo:
            pickle.dump(empleados,archivo)

    def buscaEmpleado(self, nombre: str):
        if os.path.exists(self.nomA):
            with open(self.nomA, "rb") as archivo:
                empleados = pickle.load(archivo)
                for e in empleados:
                    if e.nombre == nombre:
                        return e
        return None

    def mayorSalario(self, sueldo: float):
        if os.path.exists(self.nomA):
            with open(self.nomA, "rb") as archivo:
                empleados = pickle.load(archivo)
                for e in empleados:
                    if e.salario > sueldo:
                        return e
        return None
    

archivo = ArchivoEmpleado("empleados.dat")
archivo.crearArchivo()
archivo.guardarEmpleado(Empleado("Ana", 30, 3500.0))
archivo.guardarEmpleado(Empleado("Luis", 25, 4200.0))

print("Buscando Ana:")
print(archivo.buscaEmpleado("Ana"))

print("Buscando con salario mayor a 4000:")
print(archivo.mayorSalario(4000))

File: PERSISTENCIA/EJERCICIO-1/ArchivoEmpleado.java
package Persistencia_1;
import java.io.*;
import java.util.ArrayList;

public class ArchivoEmpleado {
    private String nomA;

    public ArchivoEmpleado(String nomA) {
        this.nomA = nomA;
    }

    public void crearArchivo() throws IOException {
        ObjectOutputStream out = new ObjectOutputStream(new FileOutputStream(nomA));
        out.writeObject(new ArrayList<Empleado>());
        out.close();
    }

    public void guardarEmpleado(Empleado e) throws IOException, ClassNotFoundException {
        ArrayList<Empleado> empleados = leerEmpleados();
        empleados.add(e);
        ObjectOutputStream out = new ObjectOutputStream(new FileOutputStream(nomA));
        out.writeObject(empleados);
        out.close();
    }

    public Empleado buscaEmpleado(String nombre) throws IOException, ClassNotFoundException {
        ArrayList<Empleado> empleados = leerEmpleados();
        for (Empleado e : empleados) {
            if (e.getNombre().equals(nombre)) {
                return e;
            }
        }
        return null;
    }

    public Empleado mayorSalario(float salario) throws IOException, ClassNotFoundException {
        ArrayList<Empleado> empleados = leerEmpleados();
        for (Empleado e : empleados) {
            if (e.getSalario() > salario) {
                return e;
            }
        }
        return null;
    }

    private ArrayList<Empleado> leerEmpleados() throws IOException, ClassNotFoundException {
        FileInputStream fileIn = new FileInputStream(nomA);
        ObjectInputStream in = new ObjectInputStream(fileIn);
        ArrayList<Empleado> empleados = (ArrayList<Empleado>) in.readObject();
        in.close();
        return empleados;
    }
}

